vscode/src/debugger/activate.ts

```diff
@@ -0,0 +1,141 @@
+import { FileAccessor, qsharpExtensionId } from "../common";
+import { QscDebugSession } from "./session";
+
+let compiler: ICompiler;
+
+export async function activateDebugger(
+  context: vscode.ExtensionContext
+): Promise<void> {
+  const workerScriptPath = vscode.Uri.joinPath(
+    context.extensionUri,
+    "./out/compilerWorker.js"
+  );
+  compiler = getCompilerWorker(workerScriptPath.toString()) as ICompiler;
+
+  registerCommands(context);
+
+  const provider = new QsDebugConfigProvider();
+  context.subscriptions.push(
+    vscode.debug.registerDebugConfigurationProvider("qsharp", provider)
+  );
+
+  const factory = new InlineDebugAdapterFactory();
+  context.subscriptions.push(
+    vscode.debug.registerDebugAdapterDescriptorFactory("qsharp", factory)
+  );
+}
+
+function registerCommands(context: vscode.ExtensionContext) {
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      `${qsharpExtensionId}.runEditorContents`,
+      (resource: vscode.Uri) => {
+        let targetResource = resource;
+        if (!targetResource && vscode.window.activeTextEditor) {
+          targetResource = vscode.window.activeTextEditor.document.uri;
+        }
+
+        if (targetResource) {
+          vscode.debug.startDebugging(
+            undefined,
+            {
+              type: "qsharp",
+              name: "Run Q# File",
+              request: "launch",
+              program: targetResource.toString(),
+              shots: 1,
+              stopOnEntry: false,
+            },
+            { noDebug: true }
+          );
+        }
+      }
+    ),
+    vscode.commands.registerCommand(
+      `${qsharpExtensionId}.debugEditorContents`,
+      (resource: vscode.Uri) => {
+        let targetResource = resource;
+        if (!targetResource && vscode.window.activeTextEditor) {
+          targetResource = vscode.window.activeTextEditor.document.uri;
+        }
+
+        if (targetResource) {
+          vscode.debug.startDebugging(undefined, {
+            type: "qsharp",
+            name: "Debug Q# File",
+            request: "launch",
+            program: targetResource,
+            shots: 1,
+            stopOnEntry: true,
+          });
+        }
+      }
+    )
+  );
+}
+
+class QsDebugConfigProvider implements vscode.DebugConfigurationProvider {
+  resolveDebugConfiguration(
+    folder: vscode.WorkspaceFolder | undefined,
+    config: vscode.DebugConfiguration,
+    _token?: vscode.CancellationToken | undefined
+  ): vscode.ProviderResult<vscode.DebugConfiguration> {
+    // if launch.json is missing or empty
+    if (!config.type && !config.request && !config.name) {
+      const editor = vscode.window.activeTextEditor;
+      if (editor && editor.document.languageId === "qsharp") {
+        config.type = "qsharp";
+        config.name = "Launch";
+        config.request = "launch";
+        config.program = editor.document.uri;
+        config.shots = 1;
```

Comment:
Do these not pick up the defaults from the package.json then?

Reply:
I didn't see the defaults coming in from the package.json for the config object when I was working on this call. Neither debug or run